### PR TITLE
Add support for Profiles

### PR DESCRIPTION
**Pull request by kabentley**
This PR combines several related-but-distinct changes that each rely on the linked native PR, so they are all included in one PR.

1. a breaking change in the @beta `CloudSqlite`  api that  adds `CloudSqite.CloudCaches` and is the new way for creating CloudCaches. `CloudContainer.detach` is now an argument to `disconnect`.
2. adds the concept of a `profile` subdirectory in `cacheDir`, supplied by the new `IModelHostOptions.profileName` parameter. All CloudCaches are within a profile directory and `IModelHost.startup` throws an exception if a second process attempt to share a profile.
3. clean up `IModelHost.startup`. Make CrashReporting @internal. 
4. a breaking change to the @alpha `CodeService` api. There are now two separate cloud databases tracking Codes for iModels, Internal and External. The internal codes database will be used to enforce uniqueness and consistency of Ids for CodeSpecs and Fonts (further work is needed on each).

imodel-native: https://github.com/iTwin/imodel-native/tree/container-methods

### CODE REVIEW

File: common/api/core-backend.api.md [hunk 1]
```diff
@@ -3325,6 +3373,16 @@ export interface InstanceChange {
     summaryId: Id64String;
 }
 
+// @internal (undocumented)
+export interface InternalCodes extends CodesDb {
+    // (undocumented)
+    reserveBisCodeSpecs(specs: [CodeService.BisCodeSpecIndexProps]): Promise<void>;
```
[the remark this answers is not in the document]
Comment (pmconne) [reply common/api/core-backend.api.md:3379]: Array (what you said it accepts): `BisCodeSpecIndexProps[]`.
Tuple of 1 (what the code declares): `[BisCodeSpecIndexProps]`.
What am I overlooking?

[the remark this answers is not in the document]
Comment (kabentley) [reply common/api/core-backend.api.md]: nothing, i had it wrong. fixed it.

**Review by pmconne — APPROVED** [review]

**Comment by grigasp** [comment]
@kabentley, with crash reporting APIs now `@internal`, is there an alternative for configuring crash reporting on our backends?

**Comment by kabentley** [comment]
not publicly, no. Who needs it other than us?

**Comment by grigasp** [comment]
> not publicly, no. Who needs it other than us?

I'm not aware of third parties, but I know GPB and our performance tests' backend use it. And I'm sure DR backend also uses it.

**Comment by kabentley** [comment]
There should be a level for "Bentley internal" for these apis that really connect to internal Bentley services, but since there isn't we used @internal to keep this stuff out of the public documentation.

**Comment by grigasp** [comment]
How should teams like DR know if it's "Bentley internal" (can use) or regular internal (shouldn't use)? 

From the core side of things, breaking internal APIs is allowed - how are we going to know that it's "Bentley internal" and the API should be treated as if it was public regarding breaking it?

IMO we need either an another doc tag to omit public API from docs or a flavor of `@internal` that says it should be treated as public (except for publishing its docs).

**Comment by kabentley** [comment]
i agree 100%. But, it will be legal to go from @internal to @bentleyinternal, but not @public to @bentleyinternal. For now, we'll have to handle this with documentation.

**Comment by kabentley** [comment]
BTW, one of the reasons for @bentleyinternal will be to "allow changes with an internal announcement". I think we should keep track of who's using these apis (somehow) so that changes (which will hopefully be rare) can be tested before they're completed.

**Pull request merged**